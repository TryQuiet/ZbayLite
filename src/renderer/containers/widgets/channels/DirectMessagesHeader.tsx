import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import ChannelHeader from '../../../components/widgets/channels/ChannelHeader'
import contactsSelectors from '../../../store/selectors/contacts'
// import usersSelectors from '../../../store/selectors/users'
import notificationCenterHandlers from '../../../store/handlers/notificationCenter'
import notificationCenterSlectors from '../../../store/selectors/notificationCenter'
import { notificationFilterType } from '../../../../shared/static'

export const mapStateToProps = (state, props) => {
  const contact = contactsSelectors.contact(props.contactId)(state)
  return {
    channel: {
      name: contact.username,
      address: props.contactId,
      displayableMessageLimit: 50
    },
    isRegisteredUsername: true,
    // isRegisteredUsername: usersSelectors.isRegisteredUsername(contact.username)(state),
    directMessage: true,
    mutedFlag:
      notificationCenterSlectors.contactFilterByAddress(contact.address)(
        state
      ) === notificationFilterType.MUTE
  }
}
export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      unmute: () =>
        notificationCenterHandlers.epics.setContactNotification(
          notificationFilterType.ALL_MESSAGES
        )
    },
    dispatch
  )
export default connect(mapStateToProps, mapDispatchToProps)(ChannelHeader)
