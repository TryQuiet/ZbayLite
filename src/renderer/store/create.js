import { applyMiddleware, createStore, compose } from 'redux'
import thunk from 'redux-thunk'
import promise from 'redux-promise-middleware'
import createDebounce from 'redux-debounced'

import reducers from './reducers'
import { errorsMiddleware } from './middlewares'

const composer = (enhancers) => {
  if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
      process.env.NODE_ENV === 'development') {
    return window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__(enhancers)
  }
  return compose(enhancers)
}

export default ({
  initialState
} = {}) => createStore(
  reducers,
  initialState,
  composer(applyMiddleware(...[errorsMiddleware, createDebounce(), thunk, promise()]))
)
