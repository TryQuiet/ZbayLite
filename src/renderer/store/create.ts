import { applyMiddleware, createStore } from 'redux'
import thunk from 'redux-thunk'
import promise from 'redux-promise-middleware'
import createDebounce from 'redux-debounced'
import createSagaMiddleware from 'redux-saga'
import rootSaga from '../sagas/index.saga'
import { composeWithDevTools } from 'redux-devtools-extension'

import reducers from './reducers'
import { errorsMiddleware } from './middlewares'

const sagaMiddleware = createSagaMiddleware()

export default (initialState = {}) => {
  const store = createStore(
    reducers,
    initialState,
    composeWithDevTools(
      applyMiddleware(...[errorsMiddleware, createDebounce(), sagaMiddleware, thunk, promise()])
    )
  )
  sagaMiddleware.run(rootSaga)
  return store
}
