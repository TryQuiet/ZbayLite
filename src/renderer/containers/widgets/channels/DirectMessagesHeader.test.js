import { mapStateToProps } from './DirectMessagesHeader'

import create from '../../../store/create'

describe('ChannelHeader', () => {
  let store = null
  beforeEach(() => {
    jest.clearAllMocks()
    store = create({
      contacts: {
        address123: {
          username: 'testusername'
        }
      }
    })
  })

  it('will receive right props', async () => {
    const props = mapStateToProps(store.getState(), { contactId: 'address123' })
    expect(props).toMatchSnapshot()
  })
})
