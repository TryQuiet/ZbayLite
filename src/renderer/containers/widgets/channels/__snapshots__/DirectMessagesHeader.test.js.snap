// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelHeader will receive right props 1`] = `
Object {
  "channel": Object {
    "address": "address123",
    "name": "testusername",
  },
  "directMessage": true,
  "isRegisteredUsername": false,
  "mutedFlag": false,
}
`;
